subcommands/git/cmd.go

```diff
@@ -0,0 +1,90 @@
+package git
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/foundriesio/fioctl/subcommands"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const GIT_CREDS_HELPER = "git-credential-fio"
+
+var (
+	helperPath string
+)
+
+func NewCommand() *cobra.Command {
+	path, err := exec.LookPath("git")
+	if err == nil {
+		helperPath = filepath.Dir(path)
+	}
+
+	cmd := &cobra.Command{
+		Use:   "configure-git",
+		Short: "Configure a source.foundries.io Git credential helper",
+		Long: `Configure a Git credential helper that allows Git to access
+source.foundries.io.
+
+This command will likely need to be run as root. It creates a symlink,
+git-credential-fio, in the same directory as the git client binary.
+
+NOTE: The credentials will need the "source:read-update" scope to work with Git`,
+		Run: doGitCreds,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			subcommands.DieNotNil(err, "Git not found on system")
+		},
+	}
+	cmd.Flags().StringVarP(&helperPath, "creds-path", "", helperPath, "Path to install credential helper")
+	return cmd
+}
+
+func findSelf() string {
+	self := os.Args[0]
+	if !filepath.IsAbs(self) {
+		logrus.Debugf("Looking up path to %s", self)
+		var err error
+		self, err = exec.LookPath(self)
+		subcommands.DieNotNil(err)
+		self, err = filepath.Abs(self)
+		subcommands.DieNotNil(err)
+	}
+	return filepath.Clean(self)
+}
+
+func doGitCreds(cmd *cobra.Command, args []string) {
+	self := findSelf()
+
+	c := exec.Command("git", "config", "--global", "credential.https://source.foundries.io.username", "fio-oauth2")
+	subcommands.DieNotNil(c.Run())
+	c = exec.Command("git", "config", "--global", "credential.https://source.foundries.io.helper", "fio")
+	subcommands.DieNotNil(c.Run())
+
+	dst := filepath.Join(helperPath, GIT_CREDS_HELPER)
+	fmt.Println("Symlinking", self, "to", dst)
+	subcommands.DieNotNil(os.Symlink(self, dst))
```

Comment:
this would need to run as root, right? so this means you'd be configuring the helper for the root user and not the user that ran `sudo ....`?